utxo.go

```diff
@@ -0,0 +1,192 @@
+	if err != nil {
+		return nil, err
+	}
+
+	if maxAmountRecipientAddress != "" && len(changeDestinations) > 0 {
+		return nil, errors.E(errors.Invalid, "no change is generated when sending max amount,"+
+			" change destinations must not be provided")
+	}
+
+	if maxAmountRecipientAddress == "" {
+		// no change specified, generate new internal address to use as change (max amount recipient)
+		maxAmountRecipientAddress, err = nextInternalAddress()
+		if err != nil {
+			return nil, fmt.Errorf("error generating internal address to use as change: %s", err.Error())
+		}
+	}
+
+	var totalInputAmount int64
+	inputScriptSizes := make([]int, len(inputs))
+	inputScripts := make([][]byte, len(inputs))
+	for i, input := range inputs {
+		totalInputAmount += input.ValueIn
+		inputScriptSizes[i] = txsizes.RedeemP2PKHSigScriptSize
+		inputScripts[i] = input.SignatureScript
+	}
+
+	var changeScriptSize int
+	if maxAmountRecipientAddress != "" {
+		changeScriptSize, err = calculateChangeScriptSize(maxAmountRecipientAddress, tx.sourceWallet.chainParams)
+	} else {
+		changeScriptSize, err = calculateMultipleChangeScriptSize(changeDestinations, tx.sourceWallet.chainParams)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	maxSignedSize := txsizes.EstimateSerializeSize(inputScriptSizes, outputs, changeScriptSize)
+	maxRequiredFee := txrules.FeeForSerializeSize(txrules.DefaultRelayFeePerKb, maxSignedSize)
+	changeAmount := totalInputAmount - totalSendAmount - int64(maxRequiredFee)
+
+	if changeAmount < 0 {
+		return nil, errors.New(ErrInsufficientBalance)
+	}
+
+	if changeAmount != 0 && !txrules.IsDustAmount(dcrutil.Amount(changeAmount), changeScriptSize, txrules.DefaultRelayFeePerKb) {
+		if changeScriptSize > txscript.MaxScriptElementSize {
+			return nil, fmt.Errorf("script size exceed maximum bytes pushable to the stack")
+		}
+
+		if maxAmountRecipientAddress != "" {
+			singleChangeDestination := TransactionDestination{
+				Address:    maxAmountRecipientAddress,
+				AtomAmount: changeAmount,
+			}
+			changeDestinations = []TransactionDestination{singleChangeDestination}
+		}
+
+		var totalChangeAmount int64
+		for _, changeDestination := range changeDestinations {
+			changeOutput, err := txhelper.MakeTxOutput(changeDestination.Address,
+				changeDestination.AtomAmount, tx.sourceWallet.chainParams)
+			if err != nil {
+				return nil, fmt.Errorf("change address error: %v", err)
+			}
+
+			totalChangeAmount += changeOutput.Value
+			outputs = append(outputs, changeOutput)
+
+			// randomize the change output that was just added
+			changeOutputIndex := len(outputs) - 1
+			txauthor.RandomizeOutputPosition(outputs, changeOutputIndex)
+		}
+
+		if totalChangeAmount > changeAmount {
+			return nil, fmt.Errorf("total amount allocated to change addresses (%s) is higher than"+
+				" actual change amount for transaction (%s)", dcrutil.Amount(totalChangeAmount).String(),
+				dcrutil.Amount(changeAmount).String())
+		}
```

Comment:
Put this block in an else statement.
```suggestion
		} else if len(changeDestinations) > 0 {

			var totalChangeAmount int64
			for _, changeDestination := range changeDestinations {
				changeOutput, err := txhelper.MakeTxOutput(changeDestination.Address,
					changeDestination.AtomAmount, tx.sourceWallet.chainParams)
				if err != nil {
					return nil, fmt.Errorf("change address error: %v", err)
				}

				totalChangeAmount += changeOutput.Value
				outputs = append(outputs, changeOutput)

				// randomize the change output that was just added
				changeOutputIndex := len(outputs) - 1
				txauthor.RandomizeOutputPosition(outputs, changeOutputIndex)
			}

			if totalChangeAmount > changeAmount {
				return nil, fmt.Errorf("total amount allocated to change addresses (%s) is higher than"+
					" actual change amount for transaction (%s)", dcrutil.Amount(totalChangeAmount).String(),
					dcrutil.Amount(changeAmount).String())
			}
		}
```